src/main/resources/buildRunner.m

```diff
@@ -0,0 +1,37 @@
+
+% This is the MATLAB script which generates the JSON file.
+function  buildrunner(varargin)
```

Comment:
Is the plan to go with the implementation and then update use a plugin method after? I think there's a big advantage to always running `buildtool <tasks> <options>` style commands. Do we plan to release with this script or is it an incremental change?

Reply:
We  have some requirements coming up in the existing  `TaskResults` class which will help us get the results in JSON format one example is duration property which does not have getter which causes failure in jsonencode(). Also capturing the stack trace in case of failure etc. this script will be temporary workaround till we get trough other requirements.

Reply:
We can create json encoded task results in a `BuildRunnerPlugin`. I think each task result exist in the `BuildRunData` after the `runTaskGraph` method is called. I'm not so sure about the duration. I think we can get each task duration. Are we reporting entire build duration/ do we need it?

Reply:
No We are not reoprting entire buidl duration however. The Duratin of each task requires getter method and it fails if we try to use jsonencode() the class itself needs bit of tweak to support this at MATLAB plugin level.

Reply:
I understand that the task can't currently be json encoded directly, but I don't quite understand why the logic from this script can't be in a plugin after the `runTaskGraph` method. Is there data missing from the tasks at that stage that is only present in the build results?

Reply:
Thats true description is not part of it.

Reply:
I don't consider this resolved.

I think that running an actual `buildtool <tasks> <options>`  command is a more important requirement than showing task durations. As soon as we replace that command by a script we have to add plugins manually. We also lose the ability to use `-ignoreUnknownOptions` and open the door to having to deal with all version compatibilities in this script. I think there's a potential design where a "Build Options" field coming in Run Build and having that present seems important. I don't like the "here's the script we actually ran" feeling we having in Run Tests with a script this long.

Reply:
Also, I think we do have access to all the information we need in a `BuildRunnerPlugin` in the `runTaskGraph` method. For example, I see task durations when I run this a build using this plugin:
```
classdef DummyPlugin < matlab.buildtool.plugins.BuildRunnerPlugin
    methods (Access = protected)
        function runTaskGraph(plugin, pluginData)
            % Run task graph
            runTaskGraph@matlab.buildtool.plugins.BuildRunnerPlugin(plugin, pluginData);
            disp([pluginData.TaskResults.Duration]);
        end
    end
end
```

Reply:
+1. I also prefer making this a plugin rather than a wrapper script. The only downside to a plugin I see is that we can't be sure the user has the recent changes that allow plugin injection via env var. But we should be able to just gracefully not fill the results table if we cannot find the report file we expect to exist.

Reply:
> TaskResults

Hi David i think Description is what we need and its still not part of the pluginData or any of its properties.